metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/MemoryAggregator.java

```diff
@@ -0,0 +1,70 @@
+import java.util.Optional;
+
+/**
+ * Aggregate Memory metrics by calculating the minimum, maximum, and average.
+ */
+public class MemoryAggregator implements RunExecutionAggregator<MemoryMetric, Double> {
+    @Override
+    public MemoryMetric getMetricFromMetrics(Metrics metrics) {
+        return metrics.getMemory();
+    }
+
+    @Override
+    public Double getMetricFromRunExecution(RunExecution runExecution) {
+        return runExecution.getMemoryRequirementsGB();
+    }
+
+    @Override
+    public Optional<RunExecution> getWorkflowExecutionFromTaskExecutions(TaskExecutions taskExecutionsForOneWorkflowRun) {
+        final List<RunExecution> taskExecutions = taskExecutionsForOneWorkflowRun.getTaskExecutions();
+        if (taskExecutions != null && taskExecutions.stream().map(RunExecution::getMemoryRequirementsGB).allMatch(Objects::nonNull)) {
+            // Get the overall memory requirement by getting the maximum memory value used
+            final Optional<Double> maxMemoryRequirement = taskExecutions.stream()
+                    .map(RunExecution::getMemoryRequirementsGB)
+                    .filter(Objects::nonNull)
+                    .max(Double::compareTo);
+            if (maxMemoryRequirement.isPresent()) {
+                return Optional.of(new RunExecution().memoryRequirementsGB(maxMemoryRequirement.get()));
+            }
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<MemoryMetric> getAggregatedMetricFromWorkflowExecutions(List<RunExecution> workflowExecutions) {
+        List<Double> memoryRequirements = getNonNullMetricsFromRunExecutions(workflowExecutions);
+        if (!memoryRequirements.isEmpty()) {
+            DoubleStatistics statistics = new DoubleStatistics(memoryRequirements);
+            return Optional.of(new MemoryMetric()
+                    .minimum(statistics.getMinimum())
+                    .maximum(statistics.getMaximum())
+                    .average(statistics.getAverage())
+                    .numberOfDataPointsForAverage(statistics.getNumberOfDataPoints()));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<MemoryMetric> getAggregatedMetricsFromAggregatedMetrics(List<MemoryMetric> aggregatedMetrics) {
+        if (!aggregatedMetrics.isEmpty()) {
+            List<DoubleStatistics> statistics = aggregatedMetrics.stream()
+                    .map(metric -> new DoubleStatistics(metric.getMinimum(), metric.getMaximum(), metric.getAverage(), metric.getNumberOfDataPointsForAverage())).toList();
+            DoubleStatistics newStatistic = DoubleStatistics.createFromStatistics(statistics);
+            return Optional.of(new MemoryMetric()
+                    .minimum(newStatistic.getMinimum())
+                    .maximum(newStatistic.getMaximum())
+                    .average(newStatistic.getAverage())
+                    .numberOfDataPointsForAverage(newStatistic.getNumberOfDataPoints()));
```

Comment:
Attempted to create a helper function in `Statistics` that would do this, but it requires that the StatisticMetric subclasses (CpuMetric, MemoryMetric, ExecutionTimeMetric, CostMetric) define their inheritance structure in the openapi.yaml. I took a crack at this and it resulted in some weird side effects. It felt a little risky pursuing this any further since we're so close to the release of 1.15 and I didn't want to accidentally break anything.

Perhaps it can be an enhancement for 1.16